src/upgrade_step_from_3p2p1.py

```diff
@@ -0,0 +1,93 @@
+from src.upgrade_step import UpgradeStep
+from file_access import FileAccess
+import xml.etree.ElementTree as ElementTree
+from local_logger import LocalLogger
+
+IOC_FILENAME = "configurations\components\_base\iocs.xml"
+
+XML_TO_ADD = """\
+<ioc autostart="true" name="ALARM" restart="true" simlevel="none">
+    <macros/>
+    <pvs/>
+    <pvsets/>
+</ioc>"""
+
+
+class UpgradeStepFrom3p2p1(UpgradeStep):
+    """
+    Add the Alarm server to the _base ioc so that it autostarts
+    """
+
+    def perform(self, file_access, logger):
+        """
+        Perform the upgrade step from version 0 to 1
+
+        Args:
+            file_access (FileAccess): file access
+            logger (LocalLogger): logger
+
+        Returns: exit code 0 success; anything else fail
+
+        """
+        try:
+            ioc_file_contents = file_access.open_file(filename=IOC_FILENAME)
+        except IOError:
+            logger.error("Can not find file to modify in config.")
+            logger.error("Filename in configuration: {0}".format(IOC_FILENAME))
+            return -1
+
+        if not self._check_prerequisits_for_file(ioc_file_contents, logger):
+            return -2
+
+        modified_file_contents = self._add_alarm_ioc(ioc_file_contents)
+        logger.info("Adding ALARM ioc to autostart in {0}.".format(IOC_FILENAME))
+
+        if not self._check_final_file_contains_alarm_ioc(logger, modified_file_contents):
+            return -3
+
+        file_access.write_file(IOC_FILENAME, modified_file_contents)
+        return 0
+
+    def _check_final_file_contains_alarm_ioc(self, logger, modified_file_contents):
+        et = ElementTree.fromstringlist(modified_file_contents)
+        ns = {"ioc_ns": 'http://epics.isis.rl.ac.uk/schema/iocs/1.0'}
+        node_count = len(et.findall(r".//ioc_ns:ioc[@name='ALARM']", ns))
+        if node_count != 1:
+            # I can not see how to generate this error but it is here because it is important
+            logger.error("IOC default component file now contains contains {0} ALARM iocs it should contain exactly 1."
+                         .format(node_count))
```

Comment:
I got this error when I modified my `iocs.xml` to look like this:

```
<?xml version="1.0" ?>
<iocs xmlns="http://epics.isis.rl.ac.uk/schema/iocs/1.0" xmlns:ioc="http://epics.isis.rl.ac.uk/schema/iocs/1.0" xmlns:xi="http://www.w3.org/2001/XInclude">
    <ioc autostart="true" name="INSTETC_01" restart="true" simlevel="none"><macros/><pvs/><pvsets/></ioc>
    <ioc autostart="true" name="ISISDAE_01" restart="true" simlevel="none"><macros/><pvs/><pvsets/></ioc>	
    <!-- Above this line only change in master branch -->	
</iocs>
```

I feel like it should work with any valid XML that conforms to the schema and not rely on the formatting of the XML. In this case we're relying on the `</ioc>` tag being on it's own line.